src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,126 @@
+
+    if (name === 'query') {
+      this.setState({ query: value });
+    }
+
+    if (name === 'sortBy') {
+      this.setState({ sortBy: value });
+    }
+  };
+
+  render() {
+    const { todos, selectedUserId, onSelectUser } = this.props;
+    const { query, sortBy } = this.state;
+
+    const filteredTodos = todos
+      .filter((todo) => {
+        if (query) {
+          return (
+            todo.title !== null
+            && todo.title.toLowerCase().includes(query.toLowerCase())
+          );
+        }
 
-        <li className="TodoList__item TodoList__item--checked">
+        return todo;
+      })
+      .filter((todo) => {
+        if (sortBy === 'active') {
+          return !todo.completed;
+        }
+
+        if (sortBy === 'completed') {
+          return todo.completed;
+        }
+
+        return todo;
+      });
+
+    return (
+      <div className="TodoList">
+        <h2>Todos:</h2>
+
+        <div className="TodoList__list-container">
           <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
+            <input
+              className="TodoList__input"
+              type="text"
+              name="query"
+              value={query}
+              onChange={this.handleSearch}
+              placeholder="Search by title"
+            />
           </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
+          <select
+            className="TodoList__select"
+            name="sortBy"
+            value={sortBy}
+            onChange={this.handleSearch}
           >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+
+          <ul className="TodoList__list">
+            {filteredTodos.map((todo) => (
+              <li
+                key={todo.id}
+                className={cn(
+                  'TodoList__item',
+                  {
+                    'TodoList__item--checked': todo.completed,
+                  },
+                  {
+                    'TodoList__item--unchecked': !todo.completed,
+                  },
+                )}
+              >
+                <label htmlFor="todoStatus">
+                  <input type="checkbox" checked={todo.completed} readOnly />
+                  <p>{todo.title}</p>
+                </label>
+
+                <button
+                  className={cn('TodoList__user-button button', {
+                    'TodoList__user-button--selected':
+                      todo.userId === selectedUserId,
+                  })}
+                  type="button"
+                  onClick={() => onSelectUser(todo.userId)}
+                >
+                  {`User&nbsp;#${todo.userId}`}
```

Comment:
That's just a common space
```suggestion
                  {`User ${todo.userId}`}
```